app/assets/javascripts/discourse/tests/acceptance/search-test.js

```diff
@@ -399,15 +399,6 @@ acceptance("Search - Anonymous", function (needs) {
       exists(".search-menu .search-context"),
       "search context indicator is visible"
     );
-
-    await fillIn("#search-term", "");
```

Comment:
Couldn't find a way to make those work. 😢 

When debugging, this `fillIn` with `""` would not clear the input field...

Reply:
That's surprising 🤔

Reply:
> When debugging, this `fillIn` with `""` would not clear the input field...

I can't repro this, it clears the input field for me locally when I restore the test code lines using your PR. But the test case fails. I think the issue is that there is a small regression with the PR. When in this state: 

<img width="535" alt="image" src="https://github.com/user-attachments/assets/71d61561-2597-4330-acf8-02c3805dbd38">

If I hit `Backspace` once, nothing happens. If I hit `Backspace` a second time, then the "in this topic" filter is cleared. 

That is, this passes: 

```
    await fillIn("#search-term", "");
    await query("#search-term").focus();
    await triggerKeyEvent("#search-term", "keyup", "Backspace");
    await triggerKeyEvent("#search-term", "keyup", "Backspace");

    assert.ok(
      !exists(".search-menu .search-context"),
      "backspace resets search context"
    );
```